flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -362,6 +440,10 @@ protected KinesisDataFetcher(List<String> streams,
 		this.watermarkTracker = watermarkTracker;
 		this.kinesisProxyFactory = checkNotNull(kinesisProxyFactory);
 		this.kinesis = kinesisProxyFactory.create(configProps);
+		this.kinesisProxyV2Factory = checkNotNull(kinesisProxyV2Factory);
+		if (shouldRegisterConsumerEagerly()) {
```

Comment:
@xiaolong-sn `EAGER` should be done in the `FlinkKinesisConsumer` constructor, `LAZY` should be done here

Reply:
Made the eager registration on FlinkKinesisConsumer